firebase-test-lab/trigger_ftl_tests.py

```
@@ -0,0 +1,346 @@
+  
+  outcome_device = []
+  # Test outcome pattern 1:
+  # ┌─────────┬──────────────────────────────┬─────────────────────┐
+  # │ OUTCOME │       TEST_AXIS_VALUE        │     TEST_DETAILS    │
+  # ├─────────┼──────────────────────────────┼─────────────────────┤
+  # │ Failed  │ iphone13pro-15.2-en-portrait │ 1 test cases failed │
+  # └─────────┴──────────────────────────────┴─────────────────────┘
+  outcome_device_search = re.findall(r'│(.*?)│(.*?)│(.*?)│', result_log, re.MULTILINE | re.DOTALL)
+  for o_d in outcome_device_search:
+    if 'OUTCOME' in o_d[0]: # skip the table title
+      continue
+    outcome_device.append({"device_axis": o_d[1].strip(), "outcome": o_d[0].strip()})
+  # Test outcome pattern 2:
+  # OUTCOME: Passed
+  # TEST_AXIS_VALUE: redfin-30-en-portrait
+  # TEST_DETAILS: --
+  outcome_device_search = re.findall(r'OUTCOME:(.*?)\nTEST_AXIS_VALUE:(.*?)\nTEST_DETAILS:', result_log, re.MULTILINE | re.DOTALL)
+  for o_d in outcome_device_search:
+    outcome_device.append({"device_axis": o_d[1].strip(), "outcome": o_d[0].strip()})
+
+  while result.poll() is None:
+    time.sleep(1) # Process hasn't exited yet, let's wait some
+  # https://firebase.google.com/docs/test-lab/android/command-line#script_exit_codes
+  logging.info("Test done: %s\nReturned code: %s\n%s", ftl_cmd, result.returncode, result_log)
+  
+  test_summary =  {
+    "return_code": result.returncode,
+    "testapp_path": testapp_path,
+    "test_type": FLAGS.test_type,
+    "ftl_link": ftl_link,
+    "raw_result_link":  raw_result_link,
+    "devices": outcome_device
+  }
+  tests_result.get('apps').append(test_summary)
+
+
+def _ftl_cmd_with_arg_group(arg_group):
+  """Returns the cmd with a YAML-formatted argument file."""
+  file_path, group_name = arg_group.split(":")
+  with open(file_path, "r") as stream:
+    try:
+      all_arg_groups = yaml.safe_load(stream)
+    except yaml.YAMLError as exc:
+      logging.error(exc)
```

Do we need to re-raise the error here?

---

Probably not. Other yaml files may be usable. Plus, if there is nothing being tested, the GHA will return 1.